R/getStepwiseDivergence.R

```diff
@@ -222,3 +233,36 @@ setMethod("getTimeDivergence",
     return(x)
 
 }
+
+############################ HELPER FUNCTION(S) ################################
+
+.get_mat_from_se <- function (x, assay_name, altexp) {
+    # non-empty string
+    if(!.is_non_empty_string(assay_name)){
+        stop("'assay_name' must be a non-empty single character value.",
+             call. = FALSE)
+    }
+    if (!is.null(altexp)) {
+        x <- altExp(x, altexp)
+    }
+    mat <- NULL
+    if (assay_name %in% assayNames(x)) {
+        mat <- t(assay(x, assay_name))
+    }
+    # x has to be an instance of SingleCellExperiment to have a reducedDim slot
+    if (is(x, "SingleCellExperiment") && assay_name %in% reducedDimNames(x)) {
```

Comment:
is(x, "SingleCellExperiment") also good to check as the first thing, nothing else will make sense if this is FALSE

Reply:
if this is FALSE then the check of assay_name on reducedDimNames won't evaluate.
In case, the altExp is a SummarizedExperiment it won't have a reducedDim slot then will pass it. Will check only for those who have a reducedDim slot e.g.  SingleCellExperiment, TreeSummarizedExperiment.

Reply:
Ah right.